### PR TITLE
params: Declare Prague hardfork configuration

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

Followed the previous process from:
- Cancun: https://github.com/klaytn/klaytn/pull/1969/files
- Kaia(Dragon): https://github.com/klaytn/klaytn/pull/2114/files
- Geth(Prague): https://github.com/ethereum/go-ethereum/pull/26481/files

Please let me know if anything is missing.

```shell
> kaia.getChainConfig()
{
  ...
  ...
  istanbulCompatibleBlock: 0,
  kaiaCompatibleBlock: 0,
  kip103CompatibleBlock: 0,
  kip103ContractAddress: "0x0000000000000000000000000000000000000000",
  kip160CompatibleBlock: 0,
  kip160ContractAddress: "0x0000000000000000000000000000000000000000",
  koreCompatibleBlock: 0,
  londonCompatibleBlock: 0,
  magmaCompatibleBlock: 0,
  pragueCompatibleBlock: 0,
  randaoCompatibleBlock: 0,
  shanghaiCompatibleBlock: 0,
}
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [ ] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -230,6 +232,8 @@ type ChainConfig struct {
 	RandaoCompatibleBlock *big.Int        `json:"randaoCompatibleBlock,omitempty"` // RandaoCompatible activate block (nil = no fork)
 	RandaoRegistry        *RegistryConfig `json:"randaoRegistry,omitempty"`        // Registry initial states
 
+	PragueCompatibleBlock *big.Int `json:"pragueCompatibleBlock,omitempty"` // PragueCompatibl activate block (nil = no fork)
```
Comment (blukat29) [inline params/config.go:235]: Please place it below KaiaCompatibleBlock.

Our ChainConfigs struct is organized in the order of
- ChainID
- Mandatory hardforks
- Optional hardforks
- Other parameters

File: params/config.go [hunk 2]
```diff
@@ -428,6 +434,11 @@ func (c *ChainConfig) IsRandaoForkEnabled(num *big.Int) bool {
 	return isForked(c.RandaoCompatibleBlock, num)
 }
 
+// IsPragueForkEnabeld returns whether num is either equal to the randao block or greater.
```
Comment (blukat29) [inline params/config.go:437]:
```suggestion
// IsPragueForkEnabled returns whether num is either equal to the prague block or greater.
```

File: params/config.go [hunk 3]
```diff
@@ -490,6 +501,7 @@ func (c *ChainConfig) CheckConfigForkOrder() error {
 		{name: "cancunBlock", block: c.CancunCompatibleBlock},
 		{name: "randaoBlock", block: c.RandaoCompatibleBlock, optional: true},
 		{name: "kaiaBlock", block: c.KaiaCompatibleBlock},
+		{name: "pragueBlock", block: c.PragueCompatibleBlock, optional: true},
```
Comment (ian0371) [inline params/config.go:504]: Is this fork optional? Does it include CL as well?

Comment (hyunsooda) [reply params/config.go]: Opps, fixed at https://github.com/kaiachain/kaia/pull/105/commits/e4d328539a866d489e07238c26b9ae81ea767549 Thank you

File: params/config.go (hunk 1 above)
Comment (hyunsooda) [reply params/config.go]: Reordered. Thanks

File: params/config.go [hunk 4]
```diff
@@ -214,6 +216,7 @@ type ChainConfig struct {
 	ShanghaiCompatibleBlock  *big.Int `json:"shanghaiCompatibleBlock,omitempty"`  // ShanghaiCompatible switch block (nil = no fork, 0 already on shanghai)
 	CancunCompatibleBlock    *big.Int `json:"cancunCompatibleBlock,omitempty"`    // CancunCompatible switch block (nil = no fork, 0 already on Cancun)
 	KaiaCompatibleBlock      *big.Int `json:"kaiaCompatibleBlock,omitempty"`      // KaiaCompatible switch block (nil = no fork, 0 already on Kaia)
+	PragueCompatibleBlock    *big.Int `json:"pragueCompatibleBlock,omitempty"`    // PragueCompatibl activate block (nil = no fork)
```
Comment (ian0371) [inline params/config.go:219]:
```suggestion
	PragueCompatibleBlock    *big.Int `json:"pragueCompatibleBlock,omitempty"`    // PragueCompatible switch block (nil = no fork)
```

File: params/config.go [hunk 5]
```diff
@@ -428,6 +433,11 @@ func (c *ChainConfig) IsRandaoForkEnabled(num *big.Int) bool {
 	return isForked(c.RandaoCompatibleBlock, num)
 }
 
+// IsPragueForkEnabeld returns whether num is either equal to the prague block or greater.
+func (c *ChainConfig) IsPragueForkEnabeld(num *big.Int) bool {
```
Comment (ian0371) [inline params/config.go:437]:
```suggestion
// IsPragueForkEnabled returns whether num is either equal to the prague block or greater.
func (c *ChainConfig) IsPragueForkEnabled(num *big.Int) bool {
```

File: params/config.go (hunk 4 above)
Comment (ian0371) [reply params/config.go]: please fix `PragueCompatibl` -> `PragueCompatible` as well.

**Review by ian0371 — APPROVED** [review]

**Review by blukat29 — APPROVED** [review]

**Pull request merged**